homeassistant/components/homeassistant/scene.py

```diff
@@ -37,19 +45,65 @@
 )
 
 SCENECONFIG = namedtuple("SceneConfig", [CONF_NAME, STATES])
+_LOGGER = logging.getLogger(__name__)
 
 
 async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Set up home assistant scene entries."""
+    _process_scenes_config(hass, async_add_entities, config)
+
+    # This platform can be loaded multiple times. Only first time register serivce.
+    if hass.services.has_service(SCENE_DOMAIN, SERVICE_RELOAD):
+        return
+
+    # Store platform for later.
+    platform = entity_platform.current_platform.get()
+
+    async def reload_config(call):
+        """Reload the scene config."""
+        try:
+            conf = await conf_util.async_hass_config_yaml(hass)
+        except HomeAssistantError as err:
+            _LOGGER.error(err)
+            return None
+
+        integration = await async_get_integration(hass, SCENE_DOMAIN)
+
+        conf = await conf_util.async_process_component_config(hass, conf, integration)
+
+        if conf is None:
+            return None
+
+        if platform is None:
+            return
+
+        await platform.async_reset()
+
+        # Extract only the config for the Home Assistant platform, ignore the rest.
+        for p_type, p_platform in config_per_platform(conf, SCENE_DOMAIN):
+            if p_type != DOMAIN:
+                continue
+
+            _process_scenes_config(hass, async_add_entities, p_platform)
+
+    hass.helpers.service.async_register_admin_service(
+        SCENE_DOMAIN, SERVICE_RELOAD, reload_config
+    )
+
+
+def _process_scenes_config(hass, async_add_entities, config):
+    """Process multiple scenes and add them."""
     scene_config = config.get(STATES)
```

Comment:
`STATES` is a required config key, so the check below just checks for a possible empty list, right?

I think we should use `dict[key]` here to make that more clear.

Reply:
The key is required, but it can be an empty list. This is the case at least one at each boot as we will set up the platform from the scene integration to make sure the service is registered.